# fastqc

![image](https://user-images.githubusercontent.com/96872843/176043639-359d461a-2185-4f68-ae09-be005285ce16.png)
[Launch the workflow at the Latch console](https://console.latch.bio/explore/62939/info)
 [![Twitter URL](https://img.shields.io/twitter/url/https/twitter.com/ge_odette.svg?style=social&label=Follow%20%40ge_odette)](https://twitter.com/ge_odette)
 
 ## Basic usage

Head to the [Latch console](https://console.latch.bio/explore/63529/info) and click to add workflow to your workspace:  provide a folder with your input files and create a working directory at the latch console.<br>

If working with nanopore fast5 files, please check in the __Process FAST5 files from nanopore__ box

If you wish, you can dictate length of Kmer to look for in the Kmer content module.<br>

Note: kmers are set to a default of 7

EXample output<br>

![](https://camo.githubusercontent.com/79a3f66d3d0cdcf2b7f89006d556ce9fecb9897c0929eb18a54d7146d1c712c7/687474703a2f2f7777772e62696f696e666f726d61746963732e626162726168616d2e61632e756b2f70726f6a656374732f6661737471632f6661737471632e706e67)

Note: As at now, there are no official documents to guide interpretation of the reports<br>

This will be in future releases but you can find a detailed guide [here](https://shiltemann.github.io/wrangling-genomics/00-quality-control/index.html)
